neomodel/util.py

```diff
@@ -205,7 +230,41 @@ def rollback(self):
             self._active_transaction = None
             self._session = None
 
-    def _object_resolution(self, result_list):
+    def _object_resolution(self, object_to_resolve):
+        # For some reason, while the type of `a_result_attribute[1]`
+        # as reported by the neo4j driver is `Node` for Node-type data
+        # retrieved from the database.
+        # When the retrieved data are Relationship-Type,
+        # the returned type is `abc.[REL_LABEL]` which is however
+        # a descendant of Relationship.
+        # Consequently, the type checking was changed for both
+        # Node, Relationship objects
+        if isinstance(object_to_resolve, Node):
+            return self._NODE_CLASS_REGISTRY[
+                frozenset(object_to_resolve.labels)
+            ].inflate(object_to_resolve)
+
+        if isinstance(object_to_resolve, Relationship):
+            return self._NODE_CLASS_REGISTRY[
+                frozenset([object_to_resolve.type])
+            ].inflate(object_to_resolve)
+
+        if isinstance(object_to_resolve, NeoPath):
+            new_nodes = []
+            for node in object_to_resolve.nodes:
+                new_nodes.append(self._object_resolution(node))
+            new_relationships = []
+            for relationship in object_to_resolve.relationships:
+                new_rel = self._object_resolution(relationship)
+                print(new_rel)
```

Comment:
Please review the use of this `print` right here, probably a debugging remnant.

Reply:
@73VW Can you remove the print or resolve the conversation, whichever is right ?